README.md

```diff
@@ -1,15 +1,381 @@
+
+`web` is important here! It specifices to Heroku to receive web traffic and provide the proper HTTP routing based on what is defined in the `index.js` file.
+
+
+## You Do: Catch up (5 mins)
+
+Go ahead and add in the Procfile with the correct line of code to the root of your directory.  
+
+`add`, `commit`, and `git push heroku deploy_starter:master` again, and open the app in the browser. What do you see now?
+
+### T/P/S (10 mins)
+
+Looks like another error!
+
+![DB Error](db_error.png "Heroku Error")
+
+Do a couple more minutes of research on this error log. Then pair up and discuss what might be the issue.
+
+## Break (10/55)
+
+## Setting the Server (10 mins)
+
+Take a look at the bottom of the `index.js` file:
+
+```js
+app.listen(3001, function() {
+  console.log("app listening on port 3001")
+})
+```
+
+We need specify to Heroku *not* to use our our local server, and connect to `http://localhost:3001/`. In Rails, the default local server is `http://localhost:3000/`, but we never need to specify that, and Heroku understands to switch to its own server once we push up.  With this app, we specify what server we want to run, and we therefore need to take that further and specify what server we want run and *when*.
+
+To do this, we are going to `set` the port to work through either local or production:
+
+```js
+// on line 8 add..
+app.set("port", process.env.PORT || 3001);
+```
+
+We can then change edit the file futher so that our app knows to `listen` on the proper port:
+
+```js
+app.listen(app.get("port"), function(){
+  console.log("It's aliiive!");
+});
+
+```
+
+## You Do: Commit Again! (15 mins)
+
+Success! Your app should be up and running.. Or is it?
+
+Try the following:
+
+- run heroku open and then heroku logs immediately after
+  - what do you see?
+- Try adding a new candidate
+  - what happens?
+
+### T/P/S (10 mins)
+
+Take 5 minutes to think and do some research on this
+
+## Heroku Addons (10 mins)
+
+Heroku provides useful tools called addons that one can install to make your deployed app must easier.
+>These are similar in a sense to Rails gems, or npm modules, except they reside only in your deployed application.
+
+There are addons for optimizing your database, monitoring the performance of your app, saving errors from your app like heroku logs, and much more.
+
+Installing many of them can be simple, as we are about to prove! The most recent issue with our app is due to the fact that we have not told Heroku what database to use.  
+
+### MongoLab
+
+MongoLab is a cloud database service that easily connects your Heroku app to a protected MongoDB. There are two ways to add MongoLab to our app:
+
+1. Through the [addon catalog](https://elements.heroku.com/addons/mongolab)
+2. By simply running a heroku command in the command line
+  - `heroku addons:create mongolab`
```

Comment:
does this require a credit card? thought there was an option about specifying sandbox